src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -414,108 +415,114 @@ bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string
             var labelHistogramName = ctx.AddInitializer(labelHistogramExpanded, new long[] { _featureHistogram[0].Length, _labelHistogram.Length }, "labelHistogramExpanded");
             var learnedAbsentFeatureLogProb = ctx.AddInitializer(_absentFeaturesLogProb, new long[] { _absentFeaturesLogProb.Length, 1 }, "absentFeaturesLogProb");
 
-            var greaterOutput = ctx.AddIntermediateVariable(null, "greaterOutput", true);
+            var greaterOutput = ctx.AddIntermediateVariable(new VectorDataViewType(BooleanDataViewType.Instance, _featureHistogram[0].Length), "greaterOutput");
             var opType = "Greater";
             ctx.CreateNode(opType, new[] { featureColumn, zero }, new[] { greaterOutput }, ctx.GetNodeName(opType), "");
 
             opType = "Cast";
-            var isFeaturePresent = ctx.AddIntermediateVariable(null, "isFeaturePresent", true);
-            var node = ctx.CreateNode(opType, greaterOutput, isFeaturePresent, ctx.GetNodeName(opType), "");
+            var castOutput = ctx.AddIntermediateVariable(new VectorDataViewType(NumberDataViewType.Single, _featureHistogram[0].Length), "castOutput");
```

Comment:
>new VectorDataViewType(NumberDataViewType.Single, _featureHistogram[0].Length) [](start = 57, length = 78)

There seem to be about two or three shapes involved here. Can you please create the shapes in one place at the top and reuse the shapes? 
It would make the code more readable and make it easier to follow along?